### PR TITLE
Check if the file exists

### PR DESCRIPTION
**Pull request by timfelle**
Ensure calls to file readers actually check if the file they are supposed to read exists.

### CODE REVIEW

**Review by njansson — CHANGES_REQUESTED** [review]
To avoid potential badness on e.g. lustre filesystem at scale, I would suggest to do the inquire on one rank, and bcast the logical.




issues on lustre filesystems, I would

**Comment by timofeymukha** [comment]
Would it be better to add a `check_exists` TBP to `generic_file_t` with
```fortran
    inquire(file=this%fname, exist=file_exists)
    if (.not. file_exists) then
       call neko_error('File does not exist: '//trim(this%fname))
    end if
```
and only have `call check_exists()` in the subtypes?

File: src/io/generic_file.f90 [hunk 1]
```diff
@@ -97,5 +99,34 @@ subroutine generic_file_set_start_counter(this, n)
...
+       ! Stop if the file does not exist
+       inquire(file=this%fname, exist=file_exists)
+       if (file_exists) send_buf = 1
+
+    end if
+    call MPI_Bcast(send_buf, 1, MPI_INT, 0, NEKO_COMM, neko_mpi_ierr)
+
+    if (send_buf .eq. 0) then
+       call neko_error('File does not exist: '//trim(this%fname))
+    end if
```
Comment (njansson) [inline src/io/generic_file.f90:127]: A bit nit picky, but how about using `MPI_LOGICAL` directly?
```suggestion

    file_exists = .false.
    send_buf = 0

    if (pe_rank .eq. 0) then
       ! Stop if the file does not exist
       inquire(file=this%fname, exist=file_exists)
    end if
    call MPI_Bcast(file_exists, 1, MPI_LOGICAL, 0, NEKO_COMM, neko_mpi_ierr)

    if (.not. file_exists) then
       call neko_error('File does not exist: '//trim(this%fname))
    end if
```

Comment (timfelle) [reply src/io/generic_file.f90:127]: It is just a bad habit comming from C++ where the boolean MPI data type does not exist.

**Review by njansson — APPROVED** [review]

**Pull request merged**